src/js/components/modal.js

```diff
@@ -0,0 +1,303 @@
+   * tab event handler based on whether it is a backward or forward tab. It
+   * also listens for usage of the escape button, which triggers the close
+   * method.
+   * @param {Event} event
+   */
+  _handleKeydown(event) {
+    // Determine if keydown is occurring within the modal
+    const currentModal = event.target.closest(this.modalSelector);
+    // If not in the modal, ignore
+    if (!currentModal) return;
+
+    switch (event.keyCode) {
+      case keyCodes.tab: {
+
+        // Create an array of all the focusable elements within the modal
+        const focusList = Array.prototype.slice.call(currentModal.querySelectorAll(this.focusElements));
+        const firstFocus = focusList[0];
+        const finalFocus = focusList[focusList.length - 1];
+
+        // If they shift tab, trigger the backward tab handler, otherwise use the forward tab handler
+        event.shiftKey ? this._handleBackwardTab(firstFocus, finalFocus, event) : this._handleForwardTab(firstFocus, finalFocus, event);
+
+        break;
+      }
+      case keyCodes.escape: {
+
+        // Check that the modal is not a dialog because the user needs to make a choice to proceed
+        if (this.dialog) return;
+
+        // Check that the current modal matches the instance's
+        if (currentModal.getAttribute(this.modalAttribute) !== this.modalDataValue) return;
+
+        this.close();
+
+        if (this.openButton !== null) this.openButton.focus();
+
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * This function is used to open a modal by triggering the modalOpen custom
+   * event, setting the modal's aria-hidden to false, and adding the
+   * rvt-modal-open class to it. It also allows developers to setup a custom
+   * callback function.
+   * @param {Function} callback 
+   */
+  open(callback) {
+    // Trigger modalOpen custom event. This event is used to control the process of closing other open modals.
+    const openEvent = dispatchCustomEvent(
+      'modalOpen',
+      this.element,
+      {
+        id: this.element.dataset.modal,
+      }
+    );
+
+    if (!openEvent) return;
+
+    this.element.setAttribute('aria-hidden', 'false');
+    document.body.classList.add('rvt-modal-open');
+
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  /**
+   * This function is used to close a modal by triggering the modalClose custom
+   * event, setting the modal's aria-hidden to true, and removing the
+   * rvt-modal-open class from it. It also allows developers to setup a custom
+   * callback function.
+   * @param {Function} callback 
+   */
+  close(callback) {
+    // Trigger modalClose custom event.
+    const closeEvent = dispatchCustomEvent(
+      'modalClose',
+      this.element,
+      {
+        id: this.element.dataset.modal
+      }
+    );
+
+    if (!closeEvent) return;
+
+    this.element.setAttribute('aria-hidden', 'true');
```

Comment:
Generally speaking, we should be using the native `hidden` attribute moving forward to show hide elements. Let's update that here to match how we're writing the rest of the components. There will probably be a few other placed elsewhere in the modal where you'll need to update that.